From 96360b43c313e91f318a14f1387a549d1a380968 Mon Sep 17 00:00:00 2001
From: Ian Walls <ian.walls@bywatersolutions.com>
Date: Thu, 8 Apr 2010 13:48:18 +0000
Subject: [PATCH] Fixes bug 3770: Cannot add to public lists from detail page
Content-Type: text/plain; charset="utf-8"

---
 virtualshelves/addbybiblionumber.pl |   18 ++++++++----------
 1 files changed, 8 insertions(+), 10 deletions(-)

diff --git a/virtualshelves/addbybiblionumber.pl b/virtualshelves/addbybiblionumber.pl
index b636c12..023f15b 100755
--- a/virtualshelves/addbybiblionumber.pl
+++ b/virtualshelves/addbybiblionumber.pl
@@ -152,18 +152,16 @@ if ( $shelfnumber || ( $shelfnumber == -1 ) ) {    # the shelf already exist.
 }
 else {    # this shelf doesn't already exist.
     my $limit = 10;
-    my ($shelflist) = GetRecentShelves(1, $limit, $loggedinuser);
+    my ($shelflist);
     my @shelvesloop;
     my %shelvesloop;
-    for my $shelf ( @{ $shelflist->[0] } ) {
-        push( @shelvesloop, $shelf->{shelfnumber} );
-        $shelvesloop{$shelf->{shelfnumber}} = $shelf->{shelfname};
-    }
-    # then open shelves...
-    ($shelflist) = GetRecentShelves(3, $limit, undef);
-    for my $shelf ( @{ $shelflist->[0] } ) {
-        push( @shelvesloop, $shelf->{shelfnumber} );
-        $shelvesloop{$shelf->{shelfnumber}} = $shelf->{shelfname};
+    #grab each type of shelf, open (type 3) should not be limited by user.
+    foreach my $shelftype (1,2,3) {
+	    my ($shelflist) = GetRecentShelves($shelftype, $limit, $shelftype == 3 ? undef : $loggedinuser);
+	    for my $shelf (@{ $shelflist->[0] }) {
+		    push(@shelvesloop, $shelf->{shelfnumber});
+		    $shelvesloop{$shelf->{shelfnumber}} = $shelf->{shelfname};
+	    }
     }
 
     if(@shelvesloop gt 0){
-- 
1.5.6.5

